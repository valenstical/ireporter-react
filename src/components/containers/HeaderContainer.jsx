/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { PropTypes } from 'prop-types';
import Header from '../views/Header';
import { getUser } from '../../selectors';
import HeaderAdmin from '../views/HeaderAdmin';
/**
 * Container component for the header view
 * @export
 * @class HeaderContainer
 * @extends {Component}
 */
class HeaderContainer extends Component {
  state = {
    showMenu: false
  };

  toggleMenu = () => {
    this.setState(prevState => ({
      showMenu: !prevState.showMenu
    }));
  }

  render() {
    const { user } = this.props;
    const { showMenu } = this.state;
    return (
      user.isAdmin
        ? (
          <HeaderAdmin
      showMenu={showMenu}
      toggleMenu={this.toggleMenu}
      />
        )
        : (
          <Header
        user={user}
        showMenu={showMenu}
        toggleMenu={this.toggleMenu}
      />
        )
    );
  }
}

const mapStateToProps = createStructuredSelector(
  { user: getUser }
);

HeaderContainer.propTypes = {
  user: PropTypes.object.isRequired,
};

export default connect(mapStateToProps, null)(HeaderContainer);
