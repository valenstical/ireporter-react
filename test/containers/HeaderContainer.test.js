import React from 'react';
import setup from '../setup';
import HeaderContainer from '../../src/components/containers/HeaderContainer';


describe('<EditProfileContainer>', () => {
  it('should render without crashing', () => {
    const wrapper = setup(<HeaderContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
