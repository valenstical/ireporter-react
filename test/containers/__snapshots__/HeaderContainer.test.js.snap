// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProfileContainer> should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <HeaderContainer>
        <Header>
          <nav
            className="navigation"
          >
            <div
              className="wrapper"
            >
              <div
                className="clear-float"
              >
                <Link
                  className="nav-brand text-brand float-left"
                  replace={false}
                  to="/"
                >
                  <a
                    className="nav-brand text-brand float-left"
                    href="/"
                    onClick={[Function]}
                  >
                    iReporter
                  </a>
                </Link>
                <ul
                  className="nav-section float-right nav-shown nav-hidden transition"
                  id="nav-section"
                >
                  <li
                    data-pg-name="Nav Item"
                  >
                    <Link
                      replace={false}
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </Link>
                  </li>
                  <li
                    data-pg-name="Nav Item"
                  >
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <a
                        href="/login"
                        onClick={[Function]}
                      >
                        Sign in
                      </a>
                    </Link>
                  </li>
                  <li
                    className="btn-cta"
                    data-pg-name="Nav Item"
                  >
                    <Link
                      className="btn-brand transition"
                      replace={false}
                      to="/sign-up"
                    >
                      <a
                        className="btn-brand transition"
                        href="/sign-up"
                        onClick={[Function]}
                      >
                        Report a case
                      </a>
                    </Link>
                  </li>
                </ul>
                <button
                  className="menu-bar"
                  type="button"
                >
                  <i
                    className="fa fa-align-justify"
                  />
                </button>
              </div>
            </div>
          </nav>
        </Header>
      </HeaderContainer>
    </Router>
  </BrowserRouter>
</Provider>
`;
